### PR TITLE
[DOC] Created ZeRO-1 wiki doc and related logistics_zero1.py example script

### PR DESCRIPTION
**Pull request by shawnakdeb**
## Description ##
Added wiki documentation for ZeRO-1 along with an example script.

cc @awslabs/raf-reviewer, @zachzzc , @zhen-jia, @XinweiFu

### CODE REVIEW

File: docs/zero_1.md [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+
+In this document, we discuss ZeRO-1 and how to use it in Ratex.
+
+As models increase in size, speeding up training becomes increasingly difficult. Typically, data-distributed parallelism (DDP) is one of the simplest ways to reduce training time through parallelization. It works by separating batches of data across multiple processors to be processed independently with gradients being summed afterwords. However, training a model like so implies that each parallel worker stores a copy of the entire model locally. Due to the memory limitations of GPU’s, this poses a strong limit on the size of a model that can be trained efficiently using DDP.
+
+ZeRO is a form of data parallelism that works to reduce peak GPU memory consumption by partitioning model states across the parallel workers. By doing so it offers the potential to train models several magnitudes larger than with simple data distributed parallelism.
+
+ZeRO-1 is the first stage of ZeRO, which specifically partitions the optimizer states. Below, we discuss how to use ZeRO-1 to parallely train your model through an example.
+
+## 1. Ratex Set-Up
```
Comment (zachzzc) [inline docs/zero_1.md:14]: We can skip this section here since we have the installation guide in the QUICKSTART

**Review by zachzzc — COMMENTED** [review]
Overall LGTM

File: docs/zero_1.md (hunk 1 above)
Comment (shawnakdeb) [reply docs/zero_1.md:14]: Settled offline, leaving Ratex setup instructions to include RAF build configurations.

File: docs/zero_1.md [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+## 3. Interface
+
+The ZeRO-1 feature is activated by setting the configuration variable, zero_opt_level. An example can be seen in line 116 and 117 of logistics_zero.py.
+
+```
+# import statement from line 12 included for clarity
+from raf import distributed as dist
+
+dcfg = dist.get_config()
+dcfg.zero_opt_level = 2
```
Comment (comaniac) [inline docs/zero_1.md:157]: Should be 1?
```suggestion
dcfg.zero_opt_level = 1
```

File: docs/zero_1.md [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+```
+
+## 4. LTC IR FILEs
+
+To see the effect of ZeRO-1, we can increase the verbosity of the output.
```
Comment (comaniac) [inline docs/zero_1.md:162]:
```suggestion
To see the effect of ZeRO-1 in the IR, we can increase the verbosity of the output.
```

**Review by comaniac — APPROVED** [review]
Otherwise LGTM

**Review by zhen-jia — APPROVED** [review]
LGTM

**Comment by comaniac** [comment]
Thanks @shawnakdeb @zachzzc @zhen-jia

**Pull request merged**